### ISSUE TITLE
Additional debug logging for failed pagerduty notifications

### ISSUE THREAD

**Issue by csquire** [prometheus/alertmanager#1541]
**What did you do?**
Sent an alert with a description was was too long for the pagerduty api to accept. I didn't check other receivers to see if they would suffer from the same lack of logging.

**What did you expect to see?**
Expected to see the error response body in the AlertManager debug logs in order to troubleshoot why pagerduty was not accepting my notification.

**What did you see instead? Under which circumstances?**
Log messages stating that their was an unexpected status code, but no additional information.

**Environment**

* System information:

N/A

* Alertmanager version:

0.15.0

* Prometheus version:

2.3.1

* Alertmanager configuration file:
```
global:
  resolve_timeout: 5m
  http_config: {}
  smtp_hello: localhost
  smtp_require_tls: true
  pagerduty_url: https://events.pagerduty.com/v2/enqueue
  hipchat_api_url: https://api.hipchat.com/
  opsgenie_api_url: https://api.opsgenie.com/
  wechat_api_url: https://qyapi.weixin.qq.com/cgi-bin/
  victorops_api_url: https://alert.victorops.com/integrations/generic/20131114/alert/
route:
  receiver: test-critical
  group_by:
  - alertname
  - severity
  group_wait: 30s
  group_interval: 2m
  repeat_interval: 30m
receivers:
- name: test-critical
  pagerduty_configs:
  - send_resolved: true
    http_config: {}
    routing_key: <secret>
    url: https://events.pagerduty.com/v2/enqueue
    client: '{{ template "pagerduty.default.client" . }}'
    client_url: '{{ template "pagerduty.default.clientURL" . }}'
    description: |
      {{ range .Alerts -}}
        {{ .Annotations.description }}
      {{ end }}
    details:
      firing: '{{ template "pagerduty.default.instances" .Alerts.Firing }}'
      num_firing: '{{ .Alerts.Firing | len }}'
      num_resolved: '{{ .Alerts.Resolved | len }}'
      resolved: '{{ template "pagerduty.default.instances" .Alerts.Resolved }}'
templates: []
```

* Alert Trigger:
The description in the alert below is one character longer than the pagerduty api will allow
```
curl -H "Content-Type: application/json" -d '[{"labels":{"alertname":"test_alert","severity":"critical"}, "annotations":{"description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus a nibh vitae massa imperdiet sollicitudin. Donec hendrerit condimentum metus eu consequat. Curabitur vulputate consequat velit in sollicitudin. Nunc malesuada tortor ex, eget accumsan tortor accumsan id. In vitae imperdiet lectus. Donec malesuada libero nec varius elementum. Sed eget metus imperdiet, mattis purus in, condimentum magna. Sed eget massa turpis. Pellentesque a lorem pretium, dapibus quam quis, egestas ex. Nullam fermentum porttitor tortor id dapibus. Cras lobortis, purus ut efficitur condimentum, nibh velit bibendum ligula, tincidunt convallis tortor tortor non nisi. Donec auctor lorem eu dui laoreet consequat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus a nibh vitae massa imperdiet sollicitudin. Donec hendrerit condimentum metus eu consequat. Curabitur vulputate consequat velit in sollicitudin. Nunc malesuada tortor ex, eget accumsan tortor accumsan id. In vitae imperdiet lectus. Donec malesuada libero nec var"}}]' http://myalertmanager/api/v1/alerts
```

* Logs:
```
level=debug ts=2018-09-04T18:07:05.389695179Z caller=dispatch.go:201 component=dispatcher msg="Received alert" alert=test_alert[f094a7d][active]
level=info ts=2018-09-04T18:07:05.75631909Z caller=cluster.go:578 component=cluster msg="gossip settled; proceeding" elapsed=10.001573507slevel=debug ts=2018-09-04T18:07:35.390319844Z caller=dispatch.go:444 component=dispatcher aggrGroup="{}:{alertname=\"test_alert\", severity=\"critical\"}" msg=Flushing alerts=[test_alert[f094a7d][active]]level=debug ts=2018-09-04T18:07:50.273183668Z caller=cluster.go:278 component=cluster memberlist="2018/09/04 18:07:50 [DEBUG] memberlist: Stream connection from=10.2.4.222:43838\n"
level=debug ts=2018-09-04T18:07:50.391144774Z caller=impl.go:548 msg="Notifying PagerDuty" incident="{}:{alertname=\"test_alert\", severity=\"critical\"}" eventType=triggerlevel=debug ts=2018-09-04T18:07:50.899097807Z caller=notify.go:668 component=dispatcher msg="Notify attempt failed" attempt=1 integration=pagerduty receiver=test-critical err="unexpected status code 400"
level=error ts=2018-09-04T18:07:50.899285804Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="cancelling notify retry for \"pagerduty\" due to unrecoverable error: unexpected status code 400"
level=error ts=2018-09-04T18:07:50.899376827Z caller=dispatch.go:279 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="cancelling notify retry for \"pagerduty\" due to unrecoverable error: unexpected status code 400"

```

**Comment by mxinden** [comment]
Thanks for the detailed report. I have moved this discussion here: https://github.com/prometheus/alertmanager/pull/1540#issuecomment-418637538

Let us know what you think!

**Comment by simonpasquier** [comment]
_Copying my comment from #1540_

The [PagerDuty v2 API](https://v2.developer.pagerduty.com/v2/page/events-api-reference#!/Event/post_create_event_json) states that the response has an optional `message` field defined as "`A description of the problem, or "Event processed" if successful`". I would have no problem with the PagerDuty notifier adding this field (if present) to the error message.

**Comment by simonpasquier** [comment]
This should be fixed since [0.15.2](https://github.com/prometheus/alertmanager/blob/master/CHANGELOG.md#0152--2018-08-14).
